topk-js/src/client/collection.rs

```diff
@@ -0,0 +1,167 @@
+use crate::data::document::NapiDocument;
+use crate::data::value::Value;
+use crate::error::TopkError;
+use crate::query::query::Query;
+use napi::bindgen_prelude::*;
+use napi_derive::napi;
+use std::collections::HashMap;
+use std::sync::Arc;
+
+#[napi]
+pub struct CollectionClient {
+    collection: String,
+    client: Arc<topk_rs::Client>,
+}
+
+#[napi(string_enum = "lowercase")]
+#[derive(Debug, Clone)]
+pub enum ConsistencyLevel {
+    Indexed,
+    Strong,
+}
+
+impl From<ConsistencyLevel> for topk_protos::v1::data::ConsistencyLevel {
+    fn from(consistency_level: ConsistencyLevel) -> Self {
+        match consistency_level {
+            ConsistencyLevel::Indexed => topk_protos::v1::data::ConsistencyLevel::Indexed,
+            ConsistencyLevel::Strong => topk_protos::v1::data::ConsistencyLevel::Strong,
+        }
+    }
+}
+
+#[napi]
+impl CollectionClient {
+    pub fn new(client: Arc<topk_rs::Client>, collection: String) -> Self {
+        Self { client, collection }
+    }
+
+    // TODO: Refactor lsn to be a string
+    #[napi]
+    pub async fn get(
+        &self,
+        id: String,
+        fields: Option<Vec<String>>,
+        lsn: Option<i64>,
+        consistency: Option<ConsistencyLevel>,
+    ) -> Result<HashMap<String, Value>> {
+        let document = self
+            .client
+            .collection(&self.collection)
+            .get(
+                id,
+                fields.unwrap_or_default(),
+                lsn.map(|l| l as u64),
+                consistency.map(|c| c.into()),
+            )
+            .await
+            .map_err(TopkError::from)?;
+
+        Ok(document
+            .fields
+            .into_iter()
+            .map(|(k, v)| (k, v.into()))
+            .collect())
+    }
+
+    // TODO: Refactor lsn to be a string
+    #[napi]
+    pub async fn count(
+        &self,
+        lsn: Option<i64>,
+        consistency: Option<ConsistencyLevel>,
+    ) -> Result<i64> {
+        let query = Query::new().count();
+
+        let docs = self
+            .client
+            .collection(&self.collection)
+            .query(
```

Comment:
you can use `client.collection(...).count()` here and the entire `match doc.fields.get("_count")` block can go awauy